templates/new/web_runner.py.txt

```diff
@@ -0,0 +1,60 @@
+import json
+import logging
+import tornado.httpserver
+import tornado.ioloop
+import tornado.options
+import tornado.web
+from surround import Runner
+from stages import {project_name}Data
+
+logging.basicConfig(level=logging.INFO)
+
+class WebRunner(Runner):
+    def prepare_runner(self):
+        self.assembler.init_assembler({project_name}Data())
+        self.application = Application(self.assembler)
+
+    def prepare_data(self):
+        print("No data prep")
+
+    def run(self, is_training=False):
+        self.prepare_runner()
+        self.application.listen(8080)
+        logging.info("Server started at http://localhost:8080")
+        tornado.ioloop.IOLoop.instance().start()
+
+
+class Application(tornado.web.Application):
+    def __init__(self, assembler):
+        handlers = [
+            (r"/estimate", EstimateHandler, {{'assembler': assembler}}),
+            (r"/info", InfoHandler, {{'assembler': assembler}}),
+        ]
+        tornado.web.Application.__init__(self, handlers)
+
+
+class EstimateHandler(tornado.web.RequestHandler):
+    def initialize(self, assembler):
+        self.assembler = assembler
+
+    def post(self):
+        data = json.loads(self.request.body)
+
+        # Clean output_data on every request
+        self.assembler.surround_data.output_data = ""
+
+        # Prepare input_date for the assembler
+        self.assembler.surround_data.input_data = data["message"]
+
+        # Execute assembler
+        self.assembler.init_assembler()
```

Comment:
`init_assembler` should only be called once ever. Here it will be called for every post request so probably best to move this to below line 38.